src/test/compile-fail/regions-free-region-ordering-caller.rs

```diff
@@ -15,20 +15,16 @@
 struct Paramd<'a> { x: &'a usize }
 
 fn call2<'a, 'b>(a: &'a usize, b: &'b usize) {
-    let z: Option<&'b &'a usize> = None;
-    //~^ ERROR reference has a longer lifetime than the data it references
+    let z: Option<&'b &'a usize> = None;//~ ERROR E0623
```

Comment:
I guess that the error in this case could be a *touch* misleading -- we'll say that "data flows from `a` into `b` here" or whatever, when in fact no dataflow actually occurs. This seems kind of like an edge case though. I'm inclined to leave it as is for *now*. We may want to reconsider in the future though. Maybe worth opening an issue about at some point.

Reply:
sure.

Reply:
Done